### PR TITLE
Scene path inference for LoadSceneAttribute & BuildSceneAttribute

### PR DESCRIPTION
**Pull request by faviann**
Enhanced flexibility of LoadSceneAttribute & BuildSceneAttribute by allowing scene paths to be also fully inferred from the CallerFilePath. 

Should the value path be omitted/null/empty from the attribute, the scene path will be derived from the test file name.
 /// (e.g., `Asset/Tests/ScreenshotTest.cs` will load `./ScreenshotTest.unity`)

I was not sure if you would've preferred this to be implemented differently but I made a pull request with this feature in case this is something you would see as useful for others. Please feel free to request some changes (or just refuse).

### CODE REVIEW

File: Runtime/Attributes/BuildSceneAttribute.cs [hunk 1]
```diff
@@ -3,8 +3,10 @@
 
 using System;
 using System.Diagnostics.CodeAnalysis;
+using System.IO;
 using System.Runtime.CompilerServices;
 using NUnit.Framework;
+using UnityEditor;
```
Comment (nowsprinting) [inline Runtime/Attributes/BuildSceneAttribute.cs:9]: unnecessary?

File: Runtime/Attributes/BuildSceneAttribute.cs [hunk 2]
```diff
@@ -24,19 +26,34 @@ public class BuildSceneAttribute : NUnitAttribute
...
         /// The path starts with `Assets/` or `Packages/` or `.`.
         /// And package name using `name` instead of `displayName`, when scenes in the package.
         /// (e.g., `Packages/com.nowsprinting.test-helper/Tests/Scenes/Scene.unity`)
+        ///
+        /// If the value is omitted, the scene name will be derived from the test file name .
+        /// (e.g., `Asset/Tests/ScreenshotTest.cs` will load `Asset/Tests/ScreenshotTest.unity`)
         /// </param>
         /// <remarks>
-        /// - For the process of including a Scene not in "Scenes in Build" to a build for player, see: <see cref="TestHelper.Editor.TemporaryBuildScenesUsingInTest"/>.
+        /// - For the process of including a Scene not in "Scenes in Build" to a build for player, see: <see cref="Editor.TemporaryBuildScenesUsingInTest"/>.
```
Comment (nowsprinting) [inline Runtime/Attributes/BuildSceneAttribute.cs:38]: unnecessary change?

File: Tests/Runtime/Attributes/BuildSceneAttributeInferredTest.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+using UnityEngine;
+
+namespace TestHelper.Attributes
+{
+    /// <seealso cref="LoadSceneAttributeTest"/>
+    [TestFixture]
+    [BuildScene]
+    public class BuildSceneAttributeInferredTest
+    {
+        private const string ObjectName = "CubeInNotInScenesInBuild";
```
Comment (nowsprinting) [inline Tests/Runtime/Attributes/BuildSceneAttributeInferredTest.cs:17]: Please give it a name different from the GameObject name used in the existing BuildSceneAttributeTest.
The test will fail when run consecutively.

File: Tests/Runtime/Attributes/BuildSceneAttributeInferredTest.cs [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+using NUnit.Framework;
+using TestHelper.RuntimeInternals;
+using UnityEngine;
+
+namespace TestHelper.Attributes
+{
+    /// <seealso cref="LoadSceneAttributeTest"/>
+    [TestFixture]
+    [BuildScene]
+    public class BuildSceneAttributeInferredTest
```
Comment (nowsprinting) [inline Tests/Runtime/Attributes/BuildSceneAttributeInferredTest.cs:15]: Please add a method to the BuildSceneAttributeTest class. (Do not create this class.)
BuildSceneAttributes can be multiple specified.

**Review by nowsprinting — CHANGES_REQUESTED** [review]
Thanks, It is a great idea!

I have written a few minor comments.
Please fix them.

**Comment by nowsprinting** [comment]
@faviann san, please rebase onto v0.7.3

File: Runtime/Attributes/BuildSceneAttribute.cs (hunk 1 above)
Comment (faviann) [reply Runtime/Attributes/BuildSceneAttribute.cs]: Agreed, I wrongly committed this

File: Runtime/Attributes/BuildSceneAttribute.cs (hunk 2 above)
Comment (faviann) [reply Runtime/Attributes/BuildSceneAttribute.cs]: Agreed, I also wrongly committed this

**Comment by faviann** [comment]
I rebased on the most recent master branch. If there is anything else needed you feel free to request additional changes. 

I will also be more careful to not let silly errors slip by.

**Review by nowsprinting — APPROVED** [review]
Thank you!

**Pull request merged**